### PR TITLE
Added delete activity page

### LINKED ISSUE #84: Delete activities

### PR DESCRIPTION
**Pull request by aaroncohen73**
Fixes #84 
## Changes in this PR.

- [X] Added delete activity page
## Testing this PR.

1. Run `nosetests`
2. Log in as a site_admin (username: "admin" password: "timesync-staging")
3. Create an activity, then go to `/activities` and delete it
### Expected Output.

```
$ nosetests
............................................................................
----------------------------------------------------------------------
Ran 76 tests in 1.273s

OK
```

@osuosl/devs

### CODE REVIEW

File: tests/test_delete_activity.py [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+
+        self.ctx = app.test_request_context()
+        self.ctx.push()
+
+        self.baseurl = app.config['TIMESYNC_URL']
+
+    def tearDown(self):
+        self.ctx.pop()
+
+    def login_admin(self):
```
Comment (pop) [inline tests/test_delete_activity.py:24]: `login_admin` and `login_noadmin` could just be `login(self, username='some_default', password='some_default_password')`. It's fine as is, just a thought. Less code ~= easier to maintain.

**Comment by pop** [comment]
@aaroncohen73 **Feedback:**
- `/activites/` needs to include a `create activity` link or form if we're modeling this after some of the other endpoints.
- Tests look well constructed.
- Implementation looks good too.

+1 from me.

**Comment by mathuin** [comment]
Tests pass, rebuilt container, started server, logged in successfully.

Created an activity named `pie` with slug `pie`.

Went to `http://localhost:5000/activities`.  Newly created activity not displayed.  Typed `pie` into slug, clicked search, activity is displayed.  Clicked delete.  Got some text and two buttons -- `Yes, delete this time` and `No`.  Clicked yes, was returned to the activities page with the text `Deletion successful`.  Typed `pie` into slug again, clicked search, activity is still displayed.

Looks like it's not actually deleting the activity -- or am I confused?

**Comment by aaroncohen73** [comment]
@mathuin If you have `TESTING` set to `TRUE`, then Pymesync will be initialized in testing mode (which doesn't actually send data to a server and behaves like you described).

**Comment by mathuin** [comment]
Reran this with `TESTING = False`, got this traceback:

```
app_1  | 172.20.0.1 - - [11/Jul/2016 18:26:34] "POST /login?next=%2F HTTP/1.1" 500 -
app_1  | Traceback (most recent call last):
app_1  |   File "/usr/lib/python2.7/site-packages/flask/app.py", line 1836, in __call__
app_1  |     return self.wsgi_app(environ, start_response)
app_1  |   File "/usr/lib/python2.7/site-packages/flask/app.py", line 1820, in wsgi_app
app_1  |     response = self.make_response(self.handle_exception(e))
app_1  |   File "/usr/lib/python2.7/site-packages/flask/app.py", line 1403, in handle_exception
app_1  |     reraise(exc_type, exc_value, tb)
app_1  |   File "/usr/lib/python2.7/site-packages/flask/app.py", line 1817, in wsgi_app
app_1  |     response = self.full_dispatch_request()
app_1  |   File "/usr/lib/python2.7/site-packages/flask/app.py", line 1477, in full_dispatch_request
app_1  |     rv = self.handle_user_exception(e)
app_1  |   File "/usr/lib/python2.7/site-packages/flask/app.py", line 1381, in handle_user_exception
app_1  |     reraise(exc_type, exc_value, tb)
app_1  |   File "/usr/lib/python2.7/site-packages/flask/app.py", line 1475, in full_dispatch_request
app_1  |     rv = self.dispatch_request()
app_1  |   File "/usr/lib/python2.7/site-packages/flask/app.py", line 1461, in dispatch_request
app_1  |     return self.view_functions[rule.endpoint](**req.view_args)
app_1  |   File "/opt/code/app/views/login.py", line 25, in login
app_1  |     error_message(token)
app_1  |   File "/opt/code/app/util.py", line 61, in error_message
app_1  |     flash("Error: " + array["pymesync error"] + " - " + array["text"])
app_1  | TypeError: cannot concatenate 'str' and 'ConnectionError' objects
```

**Comment by Kennric** [comment]
Without test mode, how does the app know it's url? Settings file?

Testing: looks good, but the confirmation form button says "yes, delete this time" instead of "yes, delete this activity"

**Comment by MaraJade** [comment]
+1

**Pull request merged**